ui/components/multichain-accounts/permissions/multichain-edit-networks-page/multichain-edit-networks-page.tsx

```diff
@@ -0,0 +1,239 @@
+import React, { useContext, useState } from 'react';
+import { CaipChainId } from '@metamask/utils';
+import { useI18nContext } from '../../../../hooks/useI18nContext';
+import {
+  Box,
+  IconName,
+  ButtonIcon,
+  ButtonIconSize,
+  Checkbox,
+  IconSize,
+  ButtonPrimary,
+  ButtonPrimarySize,
+  Text,
+  Icon,
+} from '../../../component-library';
+
+import {
+  AlignItems,
+  BackgroundColor,
+  BlockSize,
+  Display,
+  FlexDirection,
+  IconColor,
+  JustifyContent,
+  TextColor,
+  TextVariant,
+} from '../../../../helpers/constants/design-system';
+import {
+  MetaMetricsEventCategory,
+  MetaMetricsEventName,
+} from '../../../../../shared/constants/metametrics';
+import { MetaMetricsContext } from '../../../../contexts/metametrics';
+import { Content, Footer, Header, Page } from '../../../multichain/pages/page';
+import { EvmAndMultichainNetworkConfigurationsWithCaipChainId } from '../../../../selectors/selectors.types';
+import { NetworkListItem } from '../../../multichain/network-list-item';
+import { CHAIN_ID_TO_NETWORK_IMAGE_URL_MAP } from '../../../../../shared/constants/network';
+
+type MultichainEditNetworksPageProps = {
+  nonTestNetworks: EvmAndMultichainNetworkConfigurationsWithCaipChainId[];
+  testNetworks: EvmAndMultichainNetworkConfigurationsWithCaipChainId[];
+  defaultSelectedChainIds: CaipChainId[];
+  onClose: () => void;
+  onSubmit: (chainIds: CaipChainId[]) => void;
+};
+
+export const MultichainEditNetworksPage: React.FC<
+  MultichainEditNetworksPageProps
+> = ({
+  nonTestNetworks,
+  testNetworks,
+  defaultSelectedChainIds,
+  onSubmit,
+  onClose,
+}) => {
+  const t = useI18nContext();
+  const trackEvent = useContext(MetaMetricsContext);
+  const allNetworks = [...nonTestNetworks, ...testNetworks];
+
+  const [selectedChainIds, setSelectedChainIds] = useState(
+    defaultSelectedChainIds,
+  );
+
+  const selectAll = () => {
+    const allNetworksChainIds = allNetworks.map(
```

Comment:
I think we could move it outside of this function with useMemo, so we don't have to run in every time selectAll is clicked.